STM32/AC/HeatCtrl/Src/HeatCtrl.c

```diff
@@ -128,3 +128,24 @@ void setPWMPin(int pin, int pwmPct, int duration_ms)
         ctx->periodBegin = HAL_GetTick();
     }
 }
+
+void adjustPWMDown()
+{
+    for(HeatCtrl *ctx = heaters; ctx < &heaters[noOfHeaters]; ctx++)
+    {
+    	if (ctx->pwmPercent >= 1)
+    	{
+    		ctx->pwmPercent -= 1;
+    		ctx->pwmDuration = MAX_DURATION;
+    	}
+    }
+}
+
+void getPWMPinPercent(int pin, uint8_t *pwmPercent)
```

Comment:
why not returning the pwmPercent as the method's return value? i.e. just wondering if there is an advantage of passing an address around vs. the value

Reply:
No advantage really, but I did realise some problems arising from doing the shift i.e. when the user specifies port 0 (meaning targeting all ports) the method will not work. Will make a fix.